File: MyApplication/app/src/main/java/com/firsttask/itransition/AccuWeatherClient.java
package com.firsttask.itransition;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface AccuWeatherClient {
    @GET("/locations/v1/cities/geoposition/search")
    Call<AccuWeatherGsonLocation> getLocationCode(@Query("apikey") String apiKey , @Query("q") String q);

    @GET("forecasts/v1/daily/1day/{location}")
    Call<DailyForecasts> getWeather(@Query("location") String location, @Query("apikey") String apiKey1);
}

File: MyApplication/app/src/main/java/com/firsttask/itransition/retrofit/AccuweatherClient.kt
package com.firsttask.itransition.retrofit

import com.firsttask.itransition.adapter.DailyForecasts
import com.firsttask.itransition.adapter.WeatherLocationGSONAdapter
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface AccuweatherClient {
    @GET("/locations/v1/cities/geoposition/search")
    fun getLocationCode(@Query("apikey") apiKey: String?, @Query("q") q: String?): Call<WeatherLocationGSONAdapter>?

    @GET("/forecasts/v1/daily/1day/{location}")
    fun getWeather(@Path("location") location: String, @Query("apikey") apiKey1: String): Call<DailyForecasts>
}